### ISSUE TITLE
Previous card showing after swiping card 

### ISSUE THREAD

**Issue by PrinceUBS** [tarasvakulka/react-native-cards-swipe#1]
can anyone help me to fix this issu

https://user-images.githubusercontent.com/75003713/144604268-fa338ea9-564f-4d5d-a3d7-3c36329fc7dd.mov

e

**Comment by PrinceUBS** [comment]
@tarasvakulka sir, if you can help me then it's great 💖

thanks in advance 🥰⚛️

**Comment by tarasvakulka** [comment]
> can anyone help me to fix this issu
> 
>  Screen.Recording.2021-12-03.at.6.08.08.PM.mov 
> e

What react-native version you have? What OS you show on video (IOS or Android) ?

**Comment by tarasvakulka** [comment]
@PrinceUBS Try to use FastImage from [react-native-fast-image](https://www.npmjs.com/package/react-native-fast-image) instead of Image from rn, it might be a fix. Probably, it look like a bug in library, when use rn version > 0.63.4, but I need more time to investigate.

**Comment by PrinceUBS** [comment]
> > can anyone help me to fix this issu
> > Screen.Recording.2021-12-03.at.6.08.08.PM.mov
> > e
> 
> What react-native version you have? What OS you show on video (IOS or Android) ?

RN: "0.66.3" & this video recorded in iOS simulator, (I tested in my android device but same issue on android also)

**Comment by tarasvakulka** [comment]
> > > can anyone help me to fix this issu
> > > Screen.Recording.2021-12-03.at.6.08.08.PM.mov
> > > e
> > 
> > 
> > What react-native version you have? What OS you show on video (IOS or Android) ?
> 
> RN: "0.66.3" & this video recorded in iOS simulator, (I tested in my android device but same issue on android also)

Thanks for info, try if possible FastImage for now, it should fix this issue.

**Comment by PrinceUBS** [comment]
> @PrinceUBS Try to use FastImage from [react-native-fast-image](https://www.npmjs.com/package/react-native-fast-image) instead of Image from rn, it might be a fix. Probably, it look like a bug in library, when use rn version > 0.63.4, but I need more time to investigate.

Yes, i tried. first time i thought it was <Image/> component issue then i replace with <FastImage/> but still previous image is flashing once after swiping card. 

To debug this i added Text inside FastImage like below & records 


```
<FastImage
    key={String(card.src).toString()}
    source={card.src}
    style={
        {
            width: ScrRes.SCR_WIDTH * .85,
            height: ScrRes.SCR_WIDTH * aspectRatio - ScrRes.SCR_HEIGHT * .25,
            overflow: 'hidden',
            backgroundColor: COLORS.WHITE,
            borderColor: COLORS.PRIMARY,
            borderWidth: 0,
            borderRadius: 20,
        }
    }
>
    <Text style={[AboutStyle.PostTXT, { backgroundColor: '#fff', }]}>card.src: {JSON.stringify(card.src)}</Text>
</FastImage>
```
same issue 😕

**Comment by PrinceUBS** [comment]
BTW This lib other features & works smoothly...😘
just faced one issue!

**Comment by tarasvakulka** [comment]
> > @PrinceUBS Try to use FastImage from [react-native-fast-image](https://www.npmjs.com/package/react-native-fast-image) instead of Image from rn, it might be a fix. Probably, it look like a bug in library, when use rn version > 0.63.4, but I need more time to investigate.
> 
> Yes, i tried. first time i thought it was `<Image/>` component issue then i replace with `<FastImage/>` but still previous image is flashing once after swiping card.
> 
> To debug this i added Text inside FastImage like below & records
> 
> ```
> <FastImage
>     key={String(card.src).toString()}
>     source={card.src}
>     style={
>         {
>             width: ScrRes.SCR_WIDTH * .85,
>             height: ScrRes.SCR_WIDTH * aspectRatio - ScrRes.SCR_HEIGHT * .25,
>             overflow: 'hidden',
>             backgroundColor: COLORS.WHITE,
>             borderColor: COLORS.PRIMARY,
>             borderWidth: 0,
>             borderRadius: 20,
>         }
>     }
> >
>     <Text style={[AboutStyle.PostTXT, { backgroundColor: '#fff', }]}>card.src: {JSON.stringify(card.src)}</Text>
> </FastImage>
> ```
> 
> same issue 😕

Could you pls remove key prop from <FastImage /> component and try again ? If it will not help, please provide link to repo with reproducible example.

**Comment by PrinceUBS** [comment]
> > > @PrinceUBS Try to use FastImage from [react-native-fast-image](https://www.npmjs.com/package/react-native-fast-image) instead of Image from rn, it might be a fix. Probably, it look like a bug in library, when use rn version > 0.63.4, but I need more time to investigate.
> > 
> > 
> > Yes, i tried. first time i thought it was `<Image/>` component issue then i replace with `<FastImage/>` but still previous image is flashing once after swiping card.
> > To debug this i added Text inside FastImage like below & records
> > ```
> > <FastImage
> >     key={String(card.src).toString()}
> >     source={card.src}
> >     style={
> >         {
> >             width: ScrRes.SCR_WIDTH * .85,
> >             height: ScrRes.SCR_WIDTH * aspectRatio - ScrRes.SCR_HEIGHT * .25,
> >             overflow: 'hidden',
> >             backgroundColor: COLORS.WHITE,
> >             borderColor: COLORS.PRIMARY,
> >             borderWidth: 0,
> >             borderRadius: 20,
> >         }
> >     }
> > >
> >     <Text style={[AboutStyle.PostTXT, { backgroundColor: '#fff', }]}>card.src: {JSON.stringify(card.src)}</Text>
> > </FastImage>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > same issue 😕
> 
> Could you pls remove key prop from component and try again ? If it will not help, please provide link to repo with reproducible example.

https://github.com/PrinceUBS/CardSwiper_LightSwitch

**Comment by PrinceUBS** [comment]
> https://github.com/PrinceUBS/CardSwiper_LightSwitch

@tarasvakulka sir here repo link!

**Comment by tarasvakulka** [comment]
> > https://github.com/PrinceUBS/CardSwiper_LightSwitch
> 
> @tarasvakulka sir here repo link!

Thank for repo link! Today I released version [1.0.5](https://www.npmjs.com/package/react-native-cards-swipe/v/1.0.5) with fix, could you please try this version and confirm that issue go away ?

**Comment by PrinceUBS** [comment]
> Thank for repo link! Today I released version [1.0.5](https://www.npmjs.com/package/react-native-cards-swipe/v/1.0.5) with fix, could you please try this version and confirm that issue go away ?

Yesss it's fixed🧁.  thank you 🥰<3

**Comment by PrinceUBS** [comment]
one more thing sir, in my app i need to navigate to details screen on card tap but it's not working. 
I tried with `<Pressable />` & `<TouchableWithoutFeedback />` but both aren't worked. any idea to archive this? 
did you block taps in "renderCard" prop? using "pointerEvents" etc!

**Comment by tarasvakulka** [comment]
> one more thing sir, in my app i need to navigate to details screen on card tap but it's not working. I tried with `<Pressable />` & `<TouchableWithoutFeedback />` but both aren't worked. any idea to archive this? did you block taps in "renderCard" prop? using "pointerEvents" etc!

Thx for report. Fixed in version [1.0.6](https://www.npmjs.com/package/react-native-cards-swipe/v/1.0.6)

**Comment by PrinceUBS** [comment]
> Thx for report. Fixed in version [1.0.6](https://www.npmjs.com/package/react-native-cards-swipe/v/1.0.6)

Yes, this is working now😇.
Thanks for your help. 😋🤗
